pandas/core/apply.py

```diff
@@ -1824,12 +1824,15 @@ def relabel_result(
             fun = [
                 com.get_callable_name(f) if not isinstance(f, str) else f for f in fun
             ]
-            col_idx_order = Index(s.index).get_indexer(fun)
-            s = s.iloc[col_idx_order]
+            col_idx_order = list(Index(s.index).get_indexer(fun))
+            col_idx_order = [i for i in col_idx_order if i != -1]
+            if col_idx_order:
+                s = s.iloc[col_idx_order]
```

Comment:
```suggestion
            col_idx_order = Index(s.index).get_indexer(fun)
            col_idx_order = col_idx_order[col_idx_order != -1]
            s = s.iloc[col_idx_order]
```

Reply:
That won't produce the expected behavior.
Take the "A" example in the docstring without the condition it will be:
```bash
foo  NaN
aab  NaN
bar  NaN
dat  NaN
```
Which is wrong.

This happens because:
-   `s` after dropping NaNs: `s = [2.0]`
-   `fun = ["max"]`
-   `col_idx_order` is determined by `Index(s.index).get_indexer(fun)`.
-   Since `s` only has one value with index `["mean"]` and `fun = ["max"]`, there is no match, so `col_idx_order = [-1]`.
-   The code `s = s.iloc[col_idx_order]` results in an empty `Series` because `-1` indicates no match, producing thr wrong behaviour.

Is this not supposed to be?

Reply:
Ah OK then you can add back the `if not col_idx_order.empty:` condition

Reply:
Can´t use empty on `list`.
I think the best to check if empty is that one is this condition.

Reply:
Sure that solution works. Thanks.